src/jit/lsra.h

```diff
@@ -909,8 +909,10 @@ class LinearScan : public LinearScanInterface
         assignPhysReg(getRegisterRecord(reg), interval);
     }
 
+    bool isUnassigned(RegRecord* regRec, LsraLocation lastLocation, Interval* currentInterval = nullptr);
     void checkAndClearInterval(RegRecord* regRec, RefPosition* spillRefPosition);
-    void unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPosition);
+    void spillAndCheckMoreReference(RegRecord* regrec, Interval* assignedInterval, RefPosition* spillrefposition);
+    void unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPosition, Interval* currentInterval = nullptr);
```

Comment:
I don't think it's a good idea to have a parameter that is used only for `_TARGET_ARM_`. Also, there is still repeated logic in this method. Instead, you should add an overload:
```
void unassignPhysReg(RegRecord* regRec, RegisterType regType);
```
It would spill `regRec` at `regRec->assignedInterval->recentRefPosition`, if any, and handle the additional spill (if any) if either `regType` or `regRec->assignedInterval` are `TYP_DOUBLE`.